lib/problems.rb

```diff
@@ -3,11 +3,114 @@
 # Time Complexity: ?
 # Space Complexity: ?
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
-end
+  # raise NotImplementedError, "Not implemented yet"
+  #create new stack array
+  stack_1 = Stack.new
+  #loop thorugh each character in the string
+  string.each_char do |char|
```

Comment:
I think you have the right idea here, but not quite there.

If the character is an open brace, parens etc.  Push it into the Stack.

Otherwise pop the last item off the stack and see if it matches the current character, if it doesn't return false.

Continue until the end of the string.